host_core/test/host_core/benchmark/kvcounter_test.exs

```diff
@@ -0,0 +1,199 @@
+defmodule HostCore.BenchmarkTest do
+  # We'd rather not run this test asynchronously because it's a benchmark. We'll get better
+  # results if this is the only test running at the time.
+  use ExUnit.Case, async: false
+
+  alias HostCore.Actors.ActorRpcServer
+  alias HostCore.Actors.ActorSupervisor
+  alias HostCore.WasmCloud.Native
+  alias HostCore.Linkdefs.Manager
+  alias HostCore.Providers.ProviderSupervisor
+
+  import HostCoreTest.Common, only: [cleanup: 2, standard_setup: 1, request_http: 2]
+
+  @kvcounter_key HostCoreTest.Constants.kvcounter_key()
+  @kvcounter_path HostCoreTest.Constants.kvcounter_path()
+
+  @httpserver_contract HostCoreTest.Constants.httpserver_contract()
+  @httpserver_link HostCoreTest.Constants.default_link()
+  @httpserver_path HostCoreTest.Constants.httpserver_path()
+  @httpserver_key HostCoreTest.Constants.httpserver_key()
+  @keyvalue_contract HostCoreTest.Constants.keyvalue_contract()
+  @redis_link HostCoreTest.Constants.default_link()
+  @redis_path HostCoreTest.Constants.redis_path()
+  @redis_key HostCoreTest.Constants.redis_key()
+
+  describe "Benchmarking actor invocations" do
+    setup :standard_setup
+
+    test "load test with kvcounter wasm32-unknown actor", %{
+      :evt_watcher => evt_watcher,
+      :hconfig => config,
+      :host_pid => pid
+    } do
+      on_exit(fn -> cleanup(pid, config) end)
+
+      # TODO: make configurable w/ benchee
+      num_actors = 25
+      parallel = [1, 10, 25, 50]
+
+      seed = config.cluster_seed
+
+      {msg, inv, port} =
+        setup_kvcounter_test(config, evt_watcher, @kvcounter_key, @kvcounter_path, num_actors)
+
+      # very noisy debug logs during bench
+      {:ok, _okay} = HTTPoison.start()
+      Logger.configure(level: :info)
+
+      test_config = %{
+        "http_kvcounter_request" => fn ->
+          {:ok, _resp} = request_http("http://localhost:#{port}/api/counter", 1)
+        end
+      }
+
+      # Run the test at a few specified levels of parallelism, allowing for some warmup time to let compute calm down
+      parallel
+      |> Enum.each(fn parallel ->
+        IO.puts("Benchmarking with #{num_actors} actors and #{parallel} parallel requests")
+
+        Benchee.run(test_config,
+          warmup: 1,
+          time: 5,
+          parallel: parallel
+          # save: [path: "./echo_wasm32_wasi_#{num_actors}_#{parallel}.txt"] # TODO: this should be how we compare
```

Comment:
There are a few places in this file that say "echo" which I think should say kvcounter